### DISCUSSION TITLE
DocumentApprovalWorkflow JSON script file

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5672] [General]
DocumentApprovalWorkflow JSON script file

**Comment by glime-ai** [comment]
Based on the context provided, it appears that you are looking for a JSON script file for a Document Approval Workflow using Elsa. Unfortunately, the context does not include a complete JSON example of such a workflow. However, I can guide you on how to structure a JSON for a basic document approval workflow based on the Elsa documentation and the snippets provided.

A document approval workflow typically includes activities such as receiving a document, sending notifications for approval, waiting for an approval signal, and then proceeding based on the outcome (approved or rejected). Here's a simplified structure of what the JSON might look like:

```json
{
  "name": "DocumentApprovalWorkflow",
  "activities": [
    {
      "type": "ReceiveDocument",
      "properties": {
        "DocumentId": {
          "expressions": {
            "Literal": "<Document ID>"
          }
        }
      }
    },
    {
      "type": "SendEmail",
      "properties": {
        "From": {
          "expressions": {
            "Literal": "workflow@acme.com"
          }
        },
        "To": {
          "expressions": {
            "Literal": "approver@acme.com"
          }
        },
        "Subject": {
          "expressions": {
            "Literal": "Document Approval Required"
          }
        },
        "Body": {
          "expressions": {
            "Literal": "Please review the attached document."
          }
        }
      }
    },
    {
      "type": "SignalReceived",
      "properties": {
        "Signal": {
          "expressions": {
            "Literal": "Approve"
          }
        }
      }
    },
    {
      "type": "IfElse",
      "properties": {
        "Condition": {
          "expressions": {
            "JavaScript": "signal === 'Approve'"
          }
        }
      },
      "activities": {
        "True": [
          {
            "type": "WriteHttpResponse",
            "properties": {
              "Content": {
                "expressions": {
                  "Literal": "<h1>Document Approved</h1>"
                }
              },
              "ContentType": {
                "expressions": {
                  "Literal": "text/html"
                }
              },
              "StatusCode": {
                "expressions": {
                  "Literal": "200"
                }
              }
            }
          }
        ],
        "False": [
          {
            "type": "WriteHttpResponse",
            "properties": {
              "Content": {
                "expressions": {
                  "Literal": "<h1>Document Rejected</h1>"
                }
              },
              "ContentType": {
                "expressions": {
                  "Literal": "text/html"
                }
              },
              "StatusCode": {
                "expressions": {
                  "Literal": "200"
                }
              }
            }
          }
        ]
      }
    }
  ]
}
```

Please note that this is a very basic example and does not include all possible features or activities you might need. You would need to replace placeholders like `<Document ID>` with actual values or expressions relevant to your workflow.

For a more detailed and specific workflow, you would need to refer to the Elsa documentation, particularly the sections on programmatic vs. designer workflows, loading workflows from JSON, and the JSON representations of workflows created via the designer. If you have a specific part of the workflow definition that you need assistance with, please provide more details so I can offer more targeted help.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
